File: app/src/main/java/com/softaai/designpatterns/java/behavioral/ChainOfResposibility/ErrorLogger.java
package com.softaai.designpatterns.java.behavioral.ChainOfResposibility;

/**
 * Created by Amol Pawar on 05-04-2019.
 * softAai Apps
 */
public class ErrorLogger extends AbstractLogger {

    public ErrorLogger(int level){
        this.level = level;
    }

    @Override
    protected void write(String message) {
        System.out.println("Error Console::Logger: " + message);
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/behavioral/ErrorLogger.kt
package com.softaai.designpatterns.kotlin.behavioral


/**
 * Created by Amol Pawar on 05-04-2019.
 * softAai Apps
 */
class ErrorLogger(level : Int): AbstractLogger() {

    init {
        this.level = level
    }

    override fun write(message: String) {
         println("Error Console::Logger : " + message)
    }
}